server/src/controllers/ClanWars.ts

```diff
@@ -360,13 +361,21 @@ export const ClanWars = {
         throw new MissingElementError(translate('missingParameters', user));
       }
 
-      // Check if one of the user's brutes is the clan master
+      // Check if one of the user's brutes has permission to select war fighters
       const userBrutes = await prisma.brute.findMany({
-        where: { userId: user.id, deletedAt: null },
-        select: { id: true, masterOfClan: { select: { id: true } } },
+        where: { userId: user.id, deletedAt: null, clanId: req.body.clan },
+        select: { id: true },
       });
 
-      if (!userBrutes.some((brute) => brute.masterOfClan?.id === req.body.clan)) {
+      if (userBrutes.length === 0) {
+        throw new ForbiddenError(translate('unauthorized', user));
+      }
+
+      const hasWarPermission = await Promise.all(
+        userBrutes.map((b) => hasPermission(prisma, b.id, req.body.clan, ClanPermission.SELECT_WAR_FIGHTERS)),
+      );
```

Comment:
It should probably be `userBrutes.some()` instead. If a player has 2 brutes in a clan, and only one has the permission, this will always fail